htdocs/product/card.php

```diff
@@ -307,6 +307,9 @@
 			$object->status             	 = GETPOST('statut');
 			$object->status_buy            = GETPOST('statut_buy');
 			$object->status_batch = GETPOST('status_batch');
+			if ($object->status_batch !== 0) {
```

Comment:
All the lines before are using 4 columns. Just do in all cases.
$object->batch_mask = GETPOST('batch_mask');
If status_batch is 0 then $object->batch_mask will have no effect "later" but we can store it if a value is set (in case of we switch the status_batch from 1 to 0 to 1, for example, it is always useless to keep data stored.